pkg/stub/handler.go

```diff
@@ -155,44 +226,45 @@ func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
 	case *corev1.Secret:
 		dockercfgSecret, _ := event.Object.(*corev1.Secret)
 
+		if dockercfgSecret.Name != v1alpha1.SamplesRegistryCredentials {
+			return nil
+		}
+
 		if h.samplesResource == nil {
 			dockercfgSecret.DeepCopyInto(&h.registrySecret)
 			return nil
 		}
 
-		if h.samplesResource.Spec.RegistryCredentials.Name == dockercfgSecret.Name {
-			err := h.manageDockerCfgSecret(event.Deleted, h.samplesResource, dockercfgSecret)
-			if err != nil {
-				return err
-			}
+		err := h.manageDockerCfgSecret(event.Deleted, h.samplesResource, dockercfgSecret)
+		if err != nil {
+			return err
 		}
 
 	case *v1alpha1.SamplesResource:
+		h.defaultResourceTimer.Stop()
```

Comment:
seems unnecessary, i'd just let the timer fire and then no-op when it sees the resource already exists.

that way you don't need to add the defaultResourceTimer to the handler struct.

Reply:
originally did that, but it was sub optimal

the competing concerns:
- from earlier in our review, we don't set `h.sampleResource` on the first event until we've completed all the processing for the event 
- the actual time it takes to process all the imagestreams and templates from disk and update etcd can take more than 5 seconds
- so I would see the event processing and the default create happening at the same time, and we would then have to deal with concurrency at that level, including deciding which object to use

This was much more seamless IMO

Reply:
with mutex wrapping the setting of h.samplesResource, getting rid of call to Stop, storage of timer